### PR TITLE
unit test

### PR DESCRIPTION
**Pull request by AbigiyaTY**
In this milestone we @Bantealem  and @AbigiyaTY 

- Add unit tests for all the methods in all the classes
image

### CODE REVIEW

**Review by rloterh — COMMENTED** [review]
Hi Team,
 
 While you made a great effort in this project, unfortunately, I cannot proceed to review your code.
 
 ## Invalid Code Review Request
You have submitted a project, whereby linter checks are not showing on your Pull Request. Kindly ensure that **all** required linters are properly configured with regard to GitHub actions, and **all** checks are passing. Kindly request another code review when this is done.

**Missing Checks:**
image
 
Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏
 
 Cheers and Happy coding!👏👏👏
 
 Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _Invalid Code Review Request **does not count** into the code reviews limit._

File: spec/person_spec.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+require_relative './../person'
+describe Person do
```
Comment (FrederickMih) [inline spec/person_spec.rb:2]: - [ ] What do you think of adding tests for the following methods below?
- add_rental
- can_use_services?
- correct_name(name validation)

By doing this, we shall be proud to have covered most of the functionalities in this class.

**Review by FrederickMih — CHANGES_REQUESTED** [review]
Hi @AbigiyaTY, @rloterh, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

Screenshot from 2022-10-20 09-38-51



## Highlights :green_circle: 

- All unit tests are passing :heavy_check_mark: 
- Unit testing best practices :heavy_check_mark: 


## Required Changes ♻️

- [ ] Based on the requirements of this project, we are supposed to add unit tests for all the classes in our codebase. Let's kindly add unit tests for App, Nameable, TrimmerDecorator, CapitalizeDecorator classes.

- [ ]  Please also endeavor to test most of the methods in your classes.

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by sja-thedude — APPROVED** [review]
## Project Approved 👍 

Hi @Bantealem, This is @sja-thedude your code reviewer.

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Highlights:
- All linters passing
- All requirements completed
- Professional readme

giphy

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**